controllers/dataplane/openstackdataplanenodeset_controller.go

```diff
@@ -436,31 +436,38 @@ func (r *OpenStackDataPlaneNodeSetReconciler) Reconcile(ctx context.Context, req
 			condition.RequestedReason, condition.SeverityInfo,
 			condition.DeploymentReadyRunningMessage)
 	} else if isDeploymentFailed {
-		Log.Info("Set NodeSet DeploymentReadyCondition false")
-		deployErrorMsg := ""
-		if err != nil {
-			deployErrorMsg = err.Error()
+		podsInterface := r.Kclient.CoreV1().Pods(instance.Namespace)
+		podsList, _err := podsInterface.List(ctx, v1.ListOptions{
+			LabelSelector: fmt.Sprintf("openstackdataplanedeployment=%s", failedDeployment),
+			FieldSelector: "status.phase=Failed",
+		})
+
+		if _err != nil {
+			Log.Error(err, "unable to retrieve list of pods for dataplane diagnostic")
+		} else {
+			for _, pod := range podsList.Items {
+				Log.Info(fmt.Sprintf("openstackansibleee job %s failed due to %s with message: %s", pod.Name, pod.Status.Reason, pod.Status.Message))
+			}
 		}
-		instance.Status.Conditions.MarkFalse(condition.DeploymentReadyCondition,
-			condition.ErrorReason, condition.SeverityError,
-			deployErrorMsg)
```

Comment:
I think we should keep the status update

Reply:
oops that was unintentional.